CMakeLists.txt

```diff
@@ -38,6 +38,12 @@ elseif(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
   endif()
 endif()
 
+include(${xgboost_SOURCE_DIR}/cmake/CheckSVEsupport.cmake)
+check_xgboost_sve_support()
+if(XGBOOST_COMPILER_HAS_ARM_SVE)
+    add_compile_definitions(XGBOOST_SVE_COMPILER_SUPPORT)
```

Comment:
We should prefer target-scoped primitives to global-scoped primitives in CMake. See https://cliutils.gitlab.io/modern-cmake/chapters/basics.html#targets-are-your-friend. I will go ahead and revise the pull request to replace `add_compile_definitions` with `target_compile_definitions`.

Reply:
> We should prefer target-scoped primitives to global-scoped primitives in CMake. See https://cliutils.gitlab.io/modern-cmake/chapters/basics.html#targets-are-your-friend. I will go ahead and revise the pull request to replace `add_compile_definitions` with `target_compile_definitions`.

Hey @hcho3, thanks for reviewing the code. Will you be making this change and pushing it?

Reply:
I have made the change and pushed it.

Reply:
I have made the change and pushed it.